client/forest-manager/src/in_memory.rs

```diff
@@ -124,6 +124,27 @@ where
 
         Ok(())
     }
+
+    fn get_files_by_user(
+        &self,
+        user: &sp_runtime::AccountId32,
+    ) -> Result<Vec<(HasherOutT<T>, FileMetadata)>, ErrorT<T>> {
+        let trie = TrieDBBuilder::<T>::new(&self.memdb, &self.root).build();
+        let mut files = Vec::new();
+        let mut trie_iter = trie
+            .iter()
+            .map_err(|_| ForestStorageError::FailedToCreateTrieIterator)?;
+
+        while let Some((_, value)) = trie_iter.next().transpose()? {
+            let metadata = FileMetadata::decode(&mut &value[..])?;
+            let file_key = metadata.file_key::<T::Hash>();
+            if metadata.owner == user.encode() {
```

Comment:
Why not encoding the user once, before the `while` loop. So that you don't execute it every time.

Reply:
Yep, good catch. Changed